File: pynanostate/hub.py
from __future__ import print_function
from nanomsg import *
import argparse

parser = argparse.ArgumentParser(description="A nanostate hub written by Python.")
parser.add_argument('--bind', '-B', dest = "addr", default = "tcp://*:15000", help = "The address to bind, which should be compatible with nanomsg, such as 'tcp://127.0.0.1:15000'. Bind to 'tcp://*:15000' by default.")

args = parser.parse_args()
addr = args.addr

sock = Socket(BUS)
sock.bind(args.addr)
sock.send_buffer_size = 1024 * 1024 * 1024
sock.recv_buffer_size = 1024 * 1024 * 1024
sock.set_int_option(SOL_SOCKET, RCVMAXSIZE, -1)
print("Socket bound to " + addr)

while True:
    buf = sock.recv()
    sock.send(buf)

sock.close()

File: nodenanostate/hub.js
var nano = require("nanomsg");
var argparse = require("argparse")

var parser = argparse.ArgumentParser({
  description: "A nanostate hub written by Node.js."
});
parser.addArgument(
    ["--bind", "-B"],
    {
      dest: "addr",
      defaultValue: "tcp://127.0.0.1:15000",
      help: "The address to bind, which should be compatible with nanomsg, such as 'tcp://127.0.0.1:15000'"
    });

var args = parser.parseArgs()
var addr = args.addr;

// Start socket hub
var sock = nano.socket("bus");
sock.bind(addr);
sock.sndbuf(1024 * 1024 * 1024);
sock.rcvbuf(1024 * 1024 * 1024);
sock.rcvmaxsize(-1);

sock.on("data", function (buf) {
  sock.send(buf);
});

console.log("Socket bound to " + addr);

